qiskit/quantum_info/operators/operator.py

```diff
@@ -22,6 +22,7 @@
 from typing import TYPE_CHECKING
 
 import numpy as np
+import scipy.linalg
```

Comment:
Question: should we import `scipy.linalg` at the top-level, or locally within the `power`/`power_if_unitary` functions (as before)?

Reply:
We've historically forced `scipy` imports to be runtime imports because the cost of initialising too many `scipy` submodules is very high, and it made `import qiskit` feel very sluggish for users (and developers).  We still have a lint against it, so chances are there'll be a lint failure if it's a top-level import.

That said, in the last couple of years, scipy have done a lot to make their imports much lazier and on-demand.  That means that `import scipy` now makes `scipy.linalg` available (but imported only on demand) whereas previously you had to manually import it.  I believe they may also have applied some of the lazy loading to subcomponents of the subpackages as well, so the imports might not be as costly as they used to be.

At any rate, we still have the lint right now, so it'll need to be a runtime import.

Reply:
Thanks! Changed in [a82b123](https://github.com/Qiskit/qiskit/pull/13319/commits/a82b123e17ef4e4878696733683a1042adc1fbe1). 

And this also answers my would be followup question of why we call `import scipy` on the top-level in `qsd.py`.